lib/openstack_query/handlers/client_side_handler.py

```diff
@@ -158,69 +159,29 @@ def _check_filter_func(
         func: FilterFunc, func_kwargs: Optional[FilterParams] = None
     ) -> Tuple[bool, str]:
         """
-        Method that checks a given function can accept a set of kwargs as arguments.
+        Method checks a given function can accept a set of kwargs as arguments - using the EAFP principle.
+        returns false and the error string if an error is raised - else true
         :param func: function to test
         :param func_kwargs: kwargs to test
         """
-
         signature = inspect.signature(func)
+        prop_param = list(signature.parameters.values())[0]
+
+        # a hack to get EAFP working - set a default value for prop just to see if filter function works
+        if prop_param.annotation == typing.Any:
+            prop_val = ""
+        elif prop_param.annotation == typing.Union[int, float]:
+            prop_val = 0
+        else:
+            prop_val = prop_param.annotation()
```

Comment:
We shouldn't be using the typing info at runtime, this feels like prop_val (and associated) should be encapsulated in a dedicated class which then uses properties to validate on the setter

Reply:
I don't understand? I'll leave this for a future PR

Reply:
Basically we shouldn't use type annotations as they're exactly that annotations

Really this should just be done in a dedicated class e.g.
```
class PropParam
   def __init__(prop_type: Type):
        self.prop_type: Type = prop_type
        self._held_val = None

   @property
   def val(self):
       return self._held_val

  @property.setter(val)
     def val(self, val):
          # cast to the set property type, this will fail with a type error if it's not correct
          self._held_val = self.prop_type(val)
```

Then all the checking is done for free, and Python gives a nice error such as `string "helllo" is not convertible to int`

Reply:
One for the future

Reply:
yea - I guess move ClientFIlter into a separate class (Same with ServerFilter) and handle checking and stuff there